src/main/scala/org/broadinstitute/hail/io/plink/PlinkLoader.scala

```diff
@@ -117,14 +116,15 @@ object PlinkLoader {
     val variantsBc = sc.broadcast(variants)
     sc.hadoopConfiguration.setInt("nSamples", nSamples)
 
-    val rdd = sc.hadoopFile(bedPath, classOf[PlinkInputFormat], classOf[LongWritable], classOf[VariantRecord[Int]],
+    val rdd = sc.hadoopFile(bedPath, classOf[PlinkInputFormat], classOf[LongWritable], classOf[PlinkDecoder],
       nPartitions.getOrElse(sc.defaultMinPartitions))
 
+    val fastKeys = rdd.map { case (_, decoder) => variantsBc.value(decoder.getKey)._1}
```

Comment:
Nobody is really using plinkimport so I wouldn't worry about this, but you could directly compute the keys from the splits without looking at the file, no?